frontend/src/metabase/visualizations/click-actions/actions/HideColumnAction/HideColumnAction.tsx

```diff
@@ -48,9 +49,18 @@ export const HideColumnAction: LegacyDrill = ({
           };
         }
 
-        return onUpdateVisualizationSettings({
-          "table.columns": columnSettingsCopy,
-        });
+        const vizSettings = { "table.columns": columnSettingsCopy };
+
+        const onUpdateVisualizationSettingsSdk = clicked?.extraData
```

Comment:
This is very questionable. I think the correct way is to make to work like drills where click actions do not know about the SDK. You might want to explore the `question` callback https://github.com/metabase/metabase/blob/7e879221af64a9c5422519bd40c596155289a847/frontend/src/metabase/querying/drills/utils/sort-drill.ts#L35 or something similar to archive this behavior.

Reply:
@ranquild Thanks! I think something similar to `question: () => question.updateSettings(settings)` should do the trick. Looks like I will have to change the `onChangeCardAndRun` prop to notify the SDK of the change though, so we can update the `question` in the SDK's `InteractiveQuestion` context accordingly. I think the SDK can implement its own `onChangeCardAndRun` and pass it down the tree.

https://github.com/metabase/metabase/blob/master/frontend/src/metabase/visualizations/components/ClickActions/ClickActionsPopover.tsx#L56

Reply:
@ranquild Looks like the `question` callback currently always run the question because of the `onChangeCardAndRun` in `performAction`, which is not ideal for hiding columns. I'll add a `shouldRunQuestion` flag to the click action, so we can update the question without running it.